bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/handler/EvccBaseThingHandler.java

```diff
@@ -0,0 +1,374 @@
+            }
+        }
+    }
+
+    private void createChannel(String thingKey, ThingBuilder builder, JsonElement value) {
+        ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, thingKey);
+        ItemTypeUnit typeUnit = getItemType(channelTypeUID);
+        String itemType = typeUnit.itemType;
+
+        if (!"Unknown".equals(itemType)) {
+            Channel channel = ChannelBuilder.create(new ChannelUID(getThing().getUID(), thingKey))
+                    .withType(channelTypeUID).withAcceptedItemType(itemType).build();
+            if (!getThing().getChannels().stream().anyMatch(c -> c.getUID().equals(channel.getUID()))) {
+                builder.withChannel(channel);
+            }
+        } else {
+            String valString = "Null";
+            if (null != value) {
+                valString = value.toString();
+            }
+            logUnknownChannelXml(thingKey, "Hint for type: " + valString);
+        }
+    }
+
+    protected ItemTypeUnit getItemType(ChannelTypeUID channelTypeUID) {
+        ChannelType channelType = channelTypeRegistry.getChannelType(channelTypeUID);
+        if (null != channelType) {
+            String itemType = channelType.getItemType();
+            if (null != itemType) {
+                Unit<?> unit = Utils.getUnitfromChannelType(itemType);
+                return new ItemTypeUnit(channelType, unit);
+            }
+        }
+        return new ItemTypeUnit(channelType, Units.ONE);
+    }
+
+    private void setItemValue(ItemTypeUnit itemTypeUnit, ChannelUID channelUID, JsonElement value) {
+        if (value.isJsonNull() || "".equals(itemTypeUnit.itemType)) {
+            return;
+        }
+        switch (itemTypeUnit.itemType) {
+            case CoreItemFactory.NUMBER:
+            case NUMBER_DIMENSIONLESS:
+            case NUMBER_ELECTRIC_CURRENT:
+            case NUMBER_EMISSION_INTENSITY:
+            case NUMBER_ENERGY:
+            case NUMBER_LENGTH:
+            case NUMBER_POWER:
+            case NUMBER_CURRENCY:
+            case NUMBER_ENERGY_PRICE:
+                Double finalValue = "%".equals(itemTypeUnit.unitHint) ? value.getAsDouble() / 100 : value.getAsDouble();
+                if (channelUID.getId().contains("capacity")) {
+                    updateState(channelUID, new QuantityType<>(finalValue, itemTypeUnit.unit.multiply(1000)));
+                } else if ("Wh".equals(itemTypeUnit.unitHint)) {
+                    updateState(channelUID, new QuantityType<>(finalValue, itemTypeUnit.unit.divide(1000)));
+                } else {
+                    updateState(channelUID, new QuantityType<>(finalValue, itemTypeUnit.unit));
+                }
+                break;
+            case NUMBER_TIME:
+                updateState(channelUID, QuantityType.valueOf(value.getAsDouble() + " s"));
+                break;
+            case CoreItemFactory.DATETIME:
+                updateState(channelUID, new DateTimeType(value.getAsString()));
```

Comment:
Does the value come with a timezone? If not, it should be added when possible.

Reply:
Here an example response:
```json
{
    ...
    "planProjectedEnd": "2025-07-25T07:00:00+02:00",
    ...
    "effectivePlanTime": "2025-07-25T05:00:00Z",
    ...
}
```
Time zone is always included.